pkg/controller/uniteddeployment/allocator.go

```diff
@@ -303,19 +327,22 @@ func (ac *elasticAllocator) validateAndCalculateMinMaxMap(replicas int32, nameTo
 			maxReplicas, _ = ParseSubsetReplicas(replicas, *subset.MaxReplicas)
 		}
 		if ac.Spec.Topology.ScheduleStrategy.IsAdaptive() {
-			unschedulable := isSubSetUnschedulable(subset.Name, nameToSubset)
+			unschedulable := isSubSetUnschedulable(subset.Name, existingSubsets)
 			// This means that in the Adaptive scheduling strategy, an unschedulable subset can only be scaled down, not scaled up.
 			if runningReplicas, ok := runningReplicasMap[subset.Name]; unschedulable && ok {
 				klog.InfoS("Assign min(runningReplicas, minReplicas/maxReplicas) for unschedulable subset",
 					"subset", subset.Name)
 				minReplicas = integer.Int32Min(runningReplicas, minReplicas)
 				maxReplicas = integer.Int32Min(runningReplicas, maxReplicas)
 			}
-			// To prevent healthy pod from being deleted
-			if runningReplicas := runningReplicasMap[subset.Name]; !unschedulable && runningReplicas > minReplicas {
-				klog.InfoS("Assign min(runningReplicas, maxReplicas) to minReplicas to avoid deleting running pods",
-					"subset", subset.Name, "minReplicas", minReplicas, "runningReplicas", runningReplicas, "maxReplicas", maxReplicas)
-				minReplicas = integer.Int32Min(runningReplicas, maxReplicas)
+			if !ac.Spec.Topology.ScheduleStrategy.ShouldReserveUnschedulablePods() {
```

Comment:
is it possible to not check ShouldReserveUnschedulablePods ? we should make validateAndCalculateMinMaxMap unaware of reserving logic, the reserving logic should be accumulated in adjustNextReplicasInReservedStrategy